client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -354,8 +354,12 @@ private void submitRetryPushData(
             batchId,
             newLoc,
             e);
-        pushDataRpcResponseCallback.onFailure(
-            new CelebornIOException(StatusCode.PUSH_DATA_CREATE_CONNECTION_FAIL_PRIMARY, e));
+        if (e instanceof InterruptedException) {
+          Thread.currentThread().interrupt();
```

Comment:
Could we maintain only the thread's interrupt state while still invoking callback.onFailure? And I've noticed some inconsistencies in how InterruptedException is handled in the Celeborn client.  In some cases it throws an exception, while in other cases it is simply ignored.

Reply:
Are you suggesting that we can invoke callback.onFailure(InterruptedException e) to interrupt the current thread? That seems reasonable. I believe that ignoring InterruptedException is unsafe, especially when we catch it without any specific handling.

Reply:
> Contributor Author

yes, better keep that behavior If there is no particular reason.